### PR TITLE
Sass 9021

### PR DESCRIPTION
**Pull request by ghousedev**
### Description
Move to backend mongo session, behind fs.  New branch to fix PR.

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

**Review by Shihabur-Atos — DISMISSED** [review]
LGTM

File: app/config/Modules.scala [hunk 1]
```diff
@@ -16,15 +16,26 @@
...
 package config
 
-import com.google.inject.AbstractModule
 import common.UUID
+import play.api.inject.Binding
+import play.api.{Configuration, Environment}
+import services._
 
-class Modules extends AbstractModule {
+class Modules extends play.api.inject.Module {
```
Comment (ansahdavid) [inline app/config/Modules.scala:24]: Can you change it to - `class Modules extends Module` and add `import play.api.inject.{Binding, Module}`

File: app/controllers/dividends/DividendsCYAController.scala [hunk 2]
```diff
@@ -159,4 +155,4 @@ class DividendsCYAController @Inject()(
       case _ => Redirect(controllers.dividends.routes.OtherUkDividendsAmountController.show(taxYear))
     }
   }
-}
+}
```
Comment (ansahdavid) [inline app/controllers/dividends/DividendsCYAController.scala:158]: Can we add a line / space to satisfy the Scalastyle formatting

File: it/test/controllers/dividends/DividendsCYAControllerISpec.scala [hunk 3]
```diff
@@ -449,318 +450,322 @@ class DividendsCYAControllerISpec extends IntegrationTest with ViewHelpers with
...
+
+        "has a status of 500" in {
+          result.header.status shouldBe INTERNAL_SERVER_ERROR
         }
       }
-    }
\ No newline at end of file
+    }
+  }
+}
```
Comment (ansahdavid) [inline it/test/controllers/dividends/DividendsCYAControllerISpec.scala:771]: Can we add a line / space to satisfy the Scalastyle formatting

File: app/services/StockDividendsSessionServiceImpl.scala [hunk 4]
```diff
@@ -136,5 +128,4 @@ class StockDividendsSessionService @Inject()(
         }
     }
   }
-
-}
+}
```
[the remark this answers is not in the document]
Comment (ghousedev) [reply app/services/StockDividendsSessionServiceImpl.scala]: Done

File: it/test/controllers/dividends/DividendsCYAControllerISpec.scala (hunk 3 above)
Comment (ghousedev) [reply it/test/controllers/dividends/DividendsCYAControllerISpec.scala]: Done

File: app/controllers/dividends/DividendsCYAController.scala (hunk 2 above)
Comment (ghousedev) [reply app/controllers/dividends/DividendsCYAController.scala]: Done

File: app/controllers/dividends/DividendsCYAController.scala [hunk 5]
```diff
@@ -36,30 +36,26 @@ import views.html.dividends.DividendsCYAView
...
   lazy val logger: Logger = Logger(this.getClass.getName)
   implicit val executionContext: ExecutionContext = mcc.executionContext
 
   //noinspection ScalaStyle
   def show(taxYear: Int): Action[AnyContent] = commonPredicates(taxYear, DIVIDENDS).async { implicit user =>
     if (appConfig.isJourneyAvailable(STOCK_DIVIDENDS)) {
-      stockDividendsSession.getAndHandle(taxYear)(errorHandler.internalServerError()) { (cya, prior) =>
-        StockDividendsCheckYourAnswersModel.getCyaModel(cya.flatMap(_.stockDividends), prior) match {
-          case Some(cyaData) => Future.successful(Redirect(routes.DividendsSummaryController.show(taxYear)))
+      stockDividendsSession.getAndHandle(taxYear)(Future.successful(errorHandler.internalServerError())) { (cya, prior) =>
```
[the remark this answers is not in the document]
Comment (ghousedev) [reply app/controllers/dividends/DividendsCYAController.scala]: Done

File: app/config/Modules.scala (hunk 1 above)
Comment (ghousedev) [reply app/config/Modules.scala]: Done

**Review by ansahdavid — DISMISSED** [review]

**Review by Shihabur-Atos — APPROVED** [review]
LGTM

**Review by ansahdavid — APPROVED** [review]

**Review by Shihabur-Atos — APPROVED** [review]
LGTM

**Review by ansahdavid — APPROVED** [review]

**Pull request merged**